pipeline.go

```diff
@@ -31,42 +31,30 @@ func NewPipeline(g generator.Generator, c consumer.Consumer, fs ...filter.Filter
 // Run the pipeline.
 func (p *Pipeline) Run() int {
 	var wg sync.WaitGroup
-	wg.Add(len(p.Filters) + 2)
+	wg.Add(3)
```

Comment:
Just a thought: do we really need to wait for everything in the wait group? Assuming all consumers are implemented correctly, the result should be determined when the consumer exits. So running the pipeline as follows should work:

```go
go p.Generator.Run(channel0, done)
go filterSet.Run(channel0, channel1)
p.Consumer.Run(channel1, done)
```